### PR TITLE
Position the post title above the excerpt in Post and ALC blocks [MAILPOET-1710]

### PR DESCRIPTION
**Pull request by pavel-mailpoet**

### CODE REVIEW

File: lib/Newsletter/Editor/PostTransformer.php [hunk 1]
```diff
@@ -76,20 +76,29 @@ private function getStructureWithLayout($post) {
...
     }
+    $title_position = isset($this->args['titlePosition']) ? $this->args['titlePosition'] : '';
 
     if ($featured_image_position === 'aboveTitle' || $featured_image_position === 'belowTitle') {
       $featured_image_position = 'centered';
     }
 
     if ($featured_image_position === 'centered') {
-      array_unshift($content, $title, $featured_image);
+      if ($title_position !== 'aboveExcerpt') {
```
Comment (JanJakes) [inline lib/Newsletter/Editor/PostTransformer.php]: Since it has both `if` and `else` it would probably make more sense not to use negative condition in the `if (...)`.

File: tests/acceptance/ConfirmTitleAlignmentSettingsInALCBlockCest.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    $post_title = 'Post for ALC newsletter testing';
+    $post_body = 'Magna primis leo neque litora quisque phasellus nunc himenaeos per, cursus porttitor rhoncus primis cubilia condimentum magna semper curabitur nibh, nunc nulla porttitor aptent aliquet dui nec accumsan quisque pharetra non pellentesque senectus hendrerit bibendum.';
+    $post = $I->cliToArray(sprintf("post create --format=json --porcelain --post_title='%s' --post_content='%s' --post_status='publish' --allow-root", $post_title, $post_body));
+    $I->cli(sprintf("media import https://dummyimage.com/600x400/000/fff.jpg --post_id=%s --title='A downloaded picture' --featured_image --allow-root", $post[0]));
+    $newsletterFactory = new Newsletter();
+    $newsletter = $newsletterFactory->withSubject($subject)->create();
+    // open the newsletter in editor
+    $I->login();
+    $I->amEditingNewsletter($newsletter->id);
+    $I->waitForText($post_title, 60);
```
Comment (JanJakes) [inline tests/acceptance/ConfirmTitleAlignmentSettingsInALCBlockCest.php]: Do we really need such long explicit timeouts in this test?

File: tests/acceptance/ConfirmTitleAlignmentSettingsInALCBlockCest.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    $I->waitForElementVisible('[data-automation-id="display_options"]');
+    //display only the post we created
+    $I->fillField('[data-automation-id="show_max_posts"]', '1');
+    $I->click('[data-automation-id="display_options"]');
+    // select above excerpt title position
+    $I->checkOption('[data-automation-id="title_above_excerpt"]');
+    // wait for xhr to finish loading
+    $I->waitForJS("return $.active > 0;", 60);
+    $I->waitForJS("return $.active == 0;", 60);
+    $I->waitForText($post_title, 60);
```
Comment (JanJakes) [inline tests/acceptance/ConfirmTitleAlignmentSettingsInALCBlockCest.php]: Would make it easier to read to add some empty lines also to the first part of the test. Looks kind of mixed now, I'd separate at least some main blocks (those with comments maybe) with empty lines.

File: tests/acceptance/ConfirmTitleAlignmentSettingsInALCBlockCest.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+class ConfirmTitleAlignmentSettingsInALCBlockCest {
+
+  function createWelcomeNewsletter(\AcceptanceTester $I) {
+    $I->wantTo('Confirm title alignment settings in ALC block');
+    // create a post and a newsletter with ALC block
+    $subject = 'ALC Newsletter';
+    $post_title = 'Post for ALC newsletter testing';
+    $post_body = 'Magna primis leo neque litora quisque phasellus nunc himenaeos per, cursus porttitor rhoncus primis cubilia condimentum magna semper curabitur nibh, nunc nulla porttitor aptent aliquet dui nec accumsan quisque pharetra non pellentesque senectus hendrerit bibendum.';
+    $post = $I->cliToArray(sprintf("post create --format=json --porcelain --post_title='%s' --post_content='%s' --post_status='publish' --allow-root", $post_title, $post_body));
+    $I->cli(sprintf("media import https://dummyimage.com/600x400/000/fff.jpg --post_id=%s --title='A downloaded picture' --featured_image --allow-root", $post[0]));
```
Comment (JanJakes) [inline tests/acceptance/ConfirmTitleAlignmentSettingsInALCBlockCest.php]: Isn't it better to add an image for the tests? This creates a dependency on external service.

File: tests/acceptance/ConfirmTitleAlignmentSettingsInALCBlockCest.php [hunk 5]
```diff
@@ -48,7 +48,7 @@ function createWelcomeNewsletter(\AcceptanceTester $I) {
     $I->waitForText($post_title, 60);
 
     // assert no vertical element with two text block is present
-    $I->cantSeeElement('.mailpoet_container_vertical > .mailpoet_text_block + .mailpoet_text_block');
+    $I->cantSeeElementInDOM('.mailpoet_container_vertical > .mailpoet_text_block + .mailpoet_text_block');
```
Comment (JanJakes) [inline tests/acceptance/ConfirmTitleAlignmentSettingsInALCBlockCest.php]: This was meant to be added to the previous commit or does it play some role in this one?

Comment (JanJakes) [reply tests/acceptance/ConfirmTitleAlignmentSettingsInALCBlockCest.php]: Actually, what this comment does? It seems to rename a test and change `automatedLatestContent` to `automatedLatestContentLayout` but why? What is the difference between those two and why we have both?

File: tests/javascript/newsletter_editor/blocks/automatedLatestContentLayout.spec.js [hunk 6]
```diff
@@ -567,14 +567,6 @@ describe('Automated latest content', function () {
...
         });
       });
-
-      it.skip('closes the sidepanel after "Done" is clicked', function () {
-        var mock = sinon.mock().once();
-        global.MailPoet.Modal.cancel = mock;
-        view.$('.mailpoet_done_editing').click();
-        mock.verify();
-        delete (global.MailPoet.Modal.cancel);
-      });
```
Comment (JanJakes) [inline tests/javascript/newsletter_editor/blocks/automatedLatestContentLayout.spec.js]: Why was this removed?

File: tests/javascript/newsletter_editor/blocks/automatedLatestContentLayout.spec.js [hunk 7]
```diff
@@ -474,6 +481,12 @@ describe('Automated latest content layout', function () {
         expect(model.get('featuredImagePosition')).to.equal(newValue);
       });
 
+      it('changes the model if featured image position changes', function () {
+        var newValue = 'aboveExcerpt';
```
Comment (JanJakes) [inline tests/javascript/newsletter_editor/blocks/automatedLatestContentLayout.spec.js]: Seems like you're changing from default value `aboveExcerpt` to nevValue `aboveExcerpt`, is that so? In that case it would be a nicer test to try the other value.

File: tests/javascript/newsletter_editor/blocks/automatedLatestContentLayout.spec.js (hunk 6 above)
Comment (pavel-mailpoet) [reply tests/javascript/newsletter_editor/blocks/automatedLatestContentLayout.spec.js]: Well, I renamed the file. I could've copied it, but I felt like there is no reason to keep the test around. We keep the file automatedLatestContent.js just for back-compatibility purposes. We don't update it with new features, we don't fix bugs in it, so I felt like there is no need for the test.

File: tests/javascript/newsletter_editor/blocks/automatedLatestContentLayout.spec.js (hunk 7 above)
Comment (pavel-mailpoet) [reply tests/javascript/newsletter_editor/blocks/automatedLatestContentLayout.spec.js]: Why do you think that? I can see the default is `abovePost`: https://github.com/mailpoet/mailpoet/pull/1852/files#diff-fd1c6f0c7468a26e792008e610774b58R69

And even if I try it, it seems to be the case: https://d.pr/i/JsxQ8z

**Comment by pavel-mailpoet** [comment]
About the wrong commit comments: you're right I added wrong code to wrong commits. My bad and I shouldn't have done that. But now it's too late to fix it. I would have to change history and I don't want to do that. I don't know if that is a good practice to update published commits. But most importantly I am too scared to do it because I have already lost my code when trying to fix some old commits.

File: tests/acceptance/ConfirmTitleAlignmentSettingsInALCBlockCest.php (hunk 5 above)
Comment (pavel-mailpoet) [reply tests/acceptance/ConfirmTitleAlignmentSettingsInALCBlockCest.php]: We only need  `automatedLatestContentLayout` and that is the actual version. We only keep `automatedLatestContent` for compatibility reasons. See this conversation: https://mailpoet.slack.com/archives/G6JE6CDBK/p1550570233025200

File: tests/javascript/newsletter_editor/blocks/automatedLatestContentLayout.spec.js (hunk 7 above)
Comment (JanJakes) [reply tests/javascript/newsletter_editor/blocks/automatedLatestContentLayout.spec.js]: I was confused by this line https://github.com/mailpoet/mailpoet/pull/1852/commits/eb0c27144bf3fb90064c9851b824056bf2dfed81#diff-8adc3011f2281cb263b761ca63086374R192, but actually that is a test as well.
